lib/models/routing-rule.js

```diff
@@ -0,0 +1,61 @@
+"use strict";
+
+class RoutingRule {
+  constructor(config) {
+    this.condition = config.Condition;
+    this.redirect = config.Redirect;
+    this.statusCode = (this.redirect && this.redirect.HttpRedirectCode) || 301;
+  }
+
+  getLocation(key, defaults) {
+    let redirectKey = key;
+
+    if (this.redirect.ReplaceKeyPrefixWith) {
+      if (!this.condition || !this.condition.KeyPrefixEquals) {
+        throw new Error(
+          "Cannot use ReplaceKeyPrefixWith without Condition.KeyPrefixEquals"
```

Comment:
I was curious about this check you added, and it doesn't look like S3 cares about the presence of a condition. Instead it will actually infinitely redirect each time the rule is evaluated, e.g. `<ReplaceKeyPrefixWith>replacement/</ReplaceKeyPrefixWith>` will cause the chain

`/test` -> `/replacement/test` -> `/replacement/replacement/test` -> ...